docs/csharp/fundamentals/tutorials/file-based-programs.md

```diff
@@ -0,0 +1,246 @@
+- Support for POSIX and Windows command-line conventions  
+- Built-in tab completion capabilities
+- Consistent parsing behavior across applications
+
+To add command line parsing capabilities, first add the `System.CommandLine` package. Add this directive after the existing package directive:
+
+:::code language="csharp" source="./snippets/file-based-programs/AsciiArt.cs" id="CommandLinePackage":::
+
+> [!IMPORTANT]
+>
+> The version `2.0.0-beta6` was the latest version when this tutorial was last updated. If there's a newer version available, use the latest version to ensure you have the latest security packages.
+
+Next, add the necessary using statements at the top of your file (after the `#!` and `#:package` directives):
+
+:::code language="csharp" source="./snippets/file-based-programs/AsciiArt.cs" id="Usings":::
+
+Define the delay option and messages argument. In command-line applications, options typically begin with `--` (double dash) and can accept arguments. The `--delay` option accepts an integer argument that specifies the delay in milliseconds. The `messagesArgument` defines how any remaining tokens after options are parsed as text. Each token becomes a separate string in the array, but text can be quoted to include multiple words in one token. For example, `"This is one message"` becomes a single token, while `This is four tokens` becomes four separate tokens. Add the following code to create the <xref:System.CommandLine.Option`1?displayProperty=nameWithType> and <xref:System.CommandLine.Argument`1?displayProperty=nameWithType> objects to represent the command line option and argument:
+
+:::code language="csharp" source="./snippets/file-based-programs/AsciiArt.cs" id="OptionArgument":::
+
+The preceding code defines the argument type for the `--delay` option, and that the arguments are an array of `string` values. This application has only one command, so you use the *root command*. Create a root command and configure it with the option and argument. Add the argument and option to the root command:
+
+:::code language="csharp" source="./snippets/file-based-programs/AsciiArt.cs" id="RootCommand":::
+
+Next, add the code to parse the command line arguments and handle any errors. This code validates the command line arguments and stores parsed arguments in the <xref:System.CommandLine.ParseResult?displayProperty=nameWithType> object:
+
+:::code language="csharp" source="./snippets/file-based-programs/AsciiArt.cs" id="ParseAndValidate":::
+
+The preceding code validates all command line arguments. If the validation fails, errors are written to the console, and the app exits.
+
+## Use parsed command line results
+
+Now, finish the app to use the parsed options and write the output. First, define a record to hold the parsed options. File-based apps can include type declarations, like records. They must be after all top-level statements and local functions. Add a `record` declaration to store the messages and the delay option value:
```

Comment:
I think it might good to mention class here to, simply because people may not be familiar with what a record is but they've probably heard of class. I don't think it needs extra explanation because it will look like a class and they should be able to pick it up.


```suggestion
Now, finish the app to use the parsed options and write the output. First, define a record to hold the parsed options. File-based apps can include type declarations, like records and classes. They must be after all top-level statements and local functions. Add a `record` declaration to store the messages and the delay option value:
```